### ISSUE TITLE
App Crashes each time after install, only iOS 9

### ISSUE THREAD

**Issue by lesvie** [SiftScience/sift-ios#55]
Each time after install app crashes on startup, it works fine on iOS 10. Please see attached screenshot:

**Steps to reproduce:** 
1) Install app and open it (it should be done automatically if you're using Xcode)
2) Close app completely 
3) Open again (not from Xcode, use device or simulator to open app again)
-> Crash

![screenshot-issue-ios9](https://cloud.githubusercontent.com/assets/1101035/24968742/0f24a660-1f63-11e7-9989-0883b32119c2.jpg)

> #0. Crashed: com.apple.main-thread
> 0  libdispatch.dylib              0x22158060 dispatch_resume$VARIANT$mp + 123
> 1  CoreFoundation                 0x22589735 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
> 2  CoreFoundation                 0x2258913f _CFXRegistrationPost + 390
> 3  CoreFoundation                 0x22588f1d ___CFXNotificationPost_block_invoke + 40
> 4  CoreFoundation                 0x225dfc6b -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1334
> 5  CoreFoundation                 0x224e9083 _CFXNotificationPost + 486
> 6  Foundation                     0x22d2a5df -[NSNotificationCenter postNotificationName:object:userInfo:] + 74
> 7  UIKit                          0x26dce565 -[UIApplication _sendWillEnterForegroundCallbacks] + 276
> 8  UIKit                          0x26e073a7 -[UIApplication _handleApplicationActivationWithScene:transitionContext:completion:] + 2270
> 9  UIKit                          0x26e06963 __88-[UIApplication _handleApplicationLifecycleEventWithScene:transitionContext:completion:]_block_invoke + 102
> 10 UIKit                          0x26e068bf -[UIApplication _handleApplicationLifecycleEventWithScene:transitionContext:completion:] + 398
> 11 UIKit                          0x26df24b3 __70-[UIApplication scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke + 134
> 12 UIKit                          0x26df21a1 -[UIApplication scene:didUpdateWithDiff:transitionContext:completion:] + 644
> 13 FrontBoardServices             0x23c5fcf5 __80-[FBSSceneImpl updater:didUpdateSettings:withDiff:transitionContext:completion:]_block_invoke_2 + 40
> 14 FrontBoardServices             0x23c7ac07 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 18
> 15 FrontBoardServices             0x23c7aab9 -[FBSSerialQueue _performNext] + 224
> 16 FrontBoardServices             0x23c7adb9 -[FBSSerialQueue _performNextFromRunLoopSource] + 48
> 17 CoreFoundation                 0x22599dff __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 14
> 18 CoreFoundation                 0x225999ed __CFRunLoopDoSources0 + 452
> 19 CoreFoundation                 0x22597d5b __CFRunLoopRun + 794
> 20 CoreFoundation                 0x224e7229 CFRunLoopRunSpecific + 520
> 21 CoreFoundation                 0x224e7015 CFRunLoopRunInMode + 108
> 22 GraphicsServices               0x23ad7ac9 GSEventRunModal + 160
> 23 UIKit                          0x26bbb189 UIApplicationMain + 144
> 24 Phrendly                       0xaeaa7 main (main.m:14)
> 25 libdispatch.dylib              0x2218f873 (Missing)

Log: [SiftScienceIOS9.txt](https://github.com/SiftScience/sift-ios/files/917215/SiftScienceIOS9.txt)

tested on iOS 9.* (9.0-9.3) with latest Sift version 0.9.4.

**Comment by quanpod** [comment]
Thanks @lesvie - acknowledging we've seen the issue.  Going to have @garylee1 take a look and he may ask for more details (appreciate the detailed bug report).

**Comment by justinjliu** [comment]
@garylee1 - thank you for looking into this. Any updates? :)

**Comment by garylee1** [comment]
@lesvie Are you able to reproduce this on our HelloSift test app? I followed your procedure using an iPhone 5s v9.1 simulator and our HelloSift app and was unable to trigger the crash.

**Comment by lesvie** [comment]
@garylee1 Unfortunately i can't reproduce this issue on your HelloSift test app

**Comment by garylee1** [comment]
@lesvie @justinjliu is this happening in production or in your dev environment? Are you importing the SDK through Carthage, CocoaPods, or by copying the source in directly?

**Comment by garylee1** [comment]
@lesvie @justinjliu – We've released v0.9.5, which hopefully fixes this. Please try it out when you get the chance and let us know what the results are. Thanks again for surfacing the crash.

**Comment by lesvie** [comment]
@garylee1 thank you for new version 
Unfortunately it crashes even with latest update but good news that i was able to reproduce this on your Demo app. 
Steps to reproduce:
Add following lines to your plist files:
```
	<key>UIBackgroundModes</key>
	<array>
		<string>audio</string>
		<string>voip</string>
	</array>
```
And then try to run app from the Xcode and click stop. After that do not press Run button from Xcode. Need to open this application from the Simulator.
Please find attached GIF below:
[how to reproduce GIF](https://cl.ly/2U252H2w2k0o)

**Comment by garylee1** [comment]
@lesvie I'm able to reproduce this as well now. Thank you for your help – this is very useful.

**Comment by lesvie** [comment]
@garylee1 you're welcome, please let me know if you'll need anything else.

**Comment by garylee1** [comment]
@lesvie I've released a new version v0.9.6 which incorporates a suspend / resume counter. It should hopefully address all crashes related to "over-resume". Thank you all for your patience.
